src/DynamoDbQueryBuilder.php

```diff
@@ -453,40 +477,31 @@ protected function getAll($columns = [], $limit = -1, $use_iterator = true)
         if ($limit === -1 && isset($this->limit)) {
             $limit = $this->limit;
         }
-
+        $this->applyScopes();
```

Comment:
yeah, actually ... that probably makes more sense.  This would allow the scopes to modify the limit as well.